package/MDAnalysis/core/topologyattrs.py

```diff
@@ -57,6 +57,35 @@
 from .. import _TOPOLOGY_ATTRS, _TOPOLOGY_TRANSPLANTS, _TOPOLOGY_ATTRNAMES
 
 
+def _check_dtype(func):
```

Comment:
This wrapper is missing the point of the issue (maybe the issue is poorly worded). We need to make sure that TopologyAttrs with dtype object (eg AtomNames) don't get integers added to their value array. I.e. "1" (a string) is fine but 1 (an int, but still a python object) isn't.